src/ir/type-updating.h

```diff
@@ -364,6 +364,11 @@ class GlobalTypeRewriter {
   // not appear, it is mapped to itself.
   void mapTypes(const TypeMap& oldToNewTypes);
 
+  // Builds new types after updating their contents using the hooks below and
+  // returns a map from the old types to the modified types. Used internally in
+  // update().
+  TypeMap rebuildTypes();
```

Comment:
If this is only useful internally, could it be private and placed down below?

Reply:
It's also used in the subclass, but I can make it protected at least.